plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestMinimalFunctionality.java

```diff
@@ -184,6 +187,31 @@ public void testLimitForSegmentQueries()
         assertQueryFails("SELECT * FROM " + TOPIC_AND_TABLE, "Segment query returned.*");
     }
 
+    @Test
+    public void testAggregationPushdown()
+    {
+        Session aggregationPushdownEnabledSession = Session.builder(getDistributedQueryRunner().getDefaultSession())
+                .setCatalogSessionProperty("pinot", "enable_aggregation_pushdown", "true")
+                .build();
+        String noGroupingsQueryTemplate = "SELECT COUNT(*), SUM(%1$s), AVG(%1$s), MIN(%1$s), MAX(%1$s) FROM " + TOPIC_AND_TABLE;
+        assertQueryFails(format(noGroupingsQueryTemplate, "long_number"), "Segment query returned.*");
```

Comment:
Why this is expected to fail? please add explanation.

(same below)

Reply:
Without aggregation pushdown enabled, the connector will query the servers directly. This is a way to test the limit for segment queries. A limit is enforced to avoid pinot servers crashing when too many large selects are run.